packages/palette/src/utils/useOnScroll.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState, RefObject } from "react"
+
+/**
+ * useOnScroll is used to track if an element is scrolled vertically or not
+ */
+export const useOnScroll = (ref: RefObject<HTMLElement>) => {
```

Comment:
I'd rather this be done with an IntersectionObserver if possible?

Reply:
I will check if it's possible 👍

Reply:
I tried but didn't succeed IntersectionObserver, from the scrollable area, even if we tweak around with `threshold` will be always marking as `isIntersecting: true`, because if we scroll we don't increase the viewing(in this case) we show more but we also hide more.
Maybe we can try to pair on it if you have time and really want to implement with IntersectionObserver 😄

Reply:
You'd just do it with a sentinel div. We do this in Force for replicating Sticky behavior: https://github.com/artsy/force/blob/master/src/v2/Components/Sticky/Sticky.tsx#L41-L68

Reply:
Unless I'm misunderstanding. This is just to turn on the shadow right?

Reply:
Oh, I will try with a sentinel div, thanks for the reference!

>Unless I'm misunderstanding. This is just to turn on the shadow right?
And off when we scroll back to the initial position

Reply:
Worked in this [commit](https://github.com/artsy/palette/pull/1076/commits/6ce31cc924738281107f5ba933ec0e7d78473b12) and in this case just observing the `isIntersecting` was enough